angular/run/routes.run.js

```diff
@@ -1,17 +1,17 @@
-export function RoutesRun($rootScope, $state, $auth) {
+export function RoutesRun($state, $transitions) {
     'ngInject';
 
+    let requiresAuthCriteria = {
+        to: ($state) => $state.data && $state.data.auth
+    };
 
-    let deregisterationCallback =  $rootScope.$on("$stateChangeStart", function(event, toState) {
-
-        if (toState.data && toState.data.auth) {
-            /*Cancel going to the authenticated state and go back to the login page*/
-            if (!$auth.isAuthenticated()) {
-                event.preventDefault();
-                return $state.go('app.login');
-            }
+    let redirectToLogin = ['$auth', '$transition$', ($auth) => {
```

Comment:
Oh okay my bad
Thanks!
but now since we don't need `$transition$` can't we remove it?

And could you also please try 1 more thing?
If it's possible to remove `'$auth', '$transition$' with this line:

```javascript
 let redirectToLogin = ($auth) => {
     'ngInject';
```

I think `ngInject` should manage the dependencies properly